### ISSUE TITLE
Coefficients for text regions search mechanism

### ISSUE THREAD

**Issue by Lezh1k** [maddevsio/go-idmatch#41]
Right now there are no optimal coefficients for OpenCV filters in processing step. 
There is some mechanism in text-extraction-coeff-finder , but it's not ideal because of bruteforce. 
Maybe we need to use some optimizations like Newton or something like that for coefficient searching. 

Need to make research how can we make optimal coefficients with some data set.

**Comment by vincentdhondt** [comment]
A general solution to the issue is not to use brute force, but gradient descent with a learning rate (stepsize) small enough not to overshoot local minima and large enough to converge in a reasonable time.  This seems counterintuitive because ANNs are complex models with large numbers of local minima, however empirical evidence has indicated that linear ANNs have most of their minima near the global minimum, meaning that gradient descent often performs “well enough” as a heuristic solution.

**Comment by Lezh1k** [comment]
Agree. 
Hardest part is to detect that we are closer to optimal coefficients here . But there is good enough distance function for this.

**Comment by Lezh1k** [comment]
Same file text-extraction-coefficients-finder.go. 

```
func compareRects(x00, y00, x01, y01, x10, y10, x11, y11 int, devX, devY float64) bool {
	return math.Abs(float64(x10-x00)) <= devX &&
		math.Abs(float64(y10-y00)) <= devY &&
		math.Abs(float64(x11-x01)) <= devX &&
		math.Abs(float64(y11-y01)) <= devY
}

func checkRegionsNewID(regions [][]image.Point, rects []frect, devX, devY float64) bool {
	count := 0
	for _, regIn := range regions {
		rect := gocv.BoundingRect(regIn)
		for _, regEt := range rects {
			if compareRects(rect.Min.X, rect.Min.Y, rect.Max.X, rect.Max.Y,
				regEt.x0, regEt.y0, regEt.x1, regEt.y1, devX, devY) {
				count++
			}
		}
	}
	return count >= len(rects) //warning!
}

```
We use these 2 functions.
